src/roles/pulp/handlers/main.yml

```diff
@@ -11,5 +11,9 @@
 
 - name: Restart pulp-worker
   ansible.builtin.systemd:
-    name: pulp-worker
+    daemon_reload: true
+    name: "pulp-worker@{{ pulp_worker_index }}.service"
     state: restarted
+  loop: "{{ range(1, pulp_worker_count | int + 1) | list }}"
+  loop_control:
+    loop_var: pulp_worker_index
```

Comment:
Instead of the `loop`, you can now use `name: pulp-worker.target` and let systemd figure out the details.